### PR TITLE
Fix bug preventing deletion of all configured permissions of an SP at once

### PR DESCRIPTION
**Pull request by ThumulaPerera**
$subject

Resolves: https://github.com/wso2/product-is/issues/13263

### CODE REVIEW

**Review by piraveena — APPROVED** [review]

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 1]
```diff
@@ -482,8 +482,16 @@ public static void updatePermissions(String applicationName, ApplicationPermissi
 
         try {
 
-            boolean exist = tenantGovReg.resourceExists(applicationNode);
-            if (!exist) {
+            boolean appNodeExists = tenantGovReg.resourceExists(applicationNode);
+
+            if (ArrayUtils.isEmpty(permissions)) { // no new permissions
```
Comment (bhagyasakalanka) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:487]: Lets follow the following format when commenting
- Start with a capital letter
- Ends with a full stop.

ex: `// No new permissions.`

Look for other places as well

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 2]
```diff
@@ -492,22 +500,9 @@ public static void updatePermissions(String applicationName, ApplicationPermissi
...
-            }
-
-            if (ArrayUtils.isEmpty(permissions)) {
-                return;
-            }
-
-            // no permission exist for the application, create new
-            if (childern == null || appNodeCollec.getChildCount() < 1) {
-
+            if (childern == null || appNodeCollec.getChildCount() < 1) { // no permissions exist for the application
```
Comment (bhagyasakalanka) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:503]: Lets update the comment

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java [hunk 3]
```diff
@@ -492,22 +500,9 @@ public static void updatePermissions(String applicationName, ApplicationPermissi
...
                 addPermission(applicationNode, permissions, tenantGovReg);
-
-            } else { // there are permission
+            } else { // there are existing permissions for the application
```
Comment (bhagyasakalanka) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:505]: Lets fix the comment format

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 1 above)
Comment (ImalshaG) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:487]: And let's make the comment a bit more descriptive and add it above this line

Comment (ThumulaPerera) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:487]: Outdated. `updatePermissions` method has been refactored in a later change and the comments are no longer applicable.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 3 above)
Comment (ThumulaPerera) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:505]: Outdated. `updatePermissions` method has been refactored in a later change and the comments are no longer applicable.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java (hunk 2 above)
Comment (ThumulaPerera) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java:503]: Outdated. `updatePermissions` method has been refactored in a later change and the comments are no longer applicable.

**Pull request merged**